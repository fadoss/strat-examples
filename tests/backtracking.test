params/backtracking

sload bt-labyrinth
srew [1] [7,7] using solve .
continue 1 .

sload bt-queens
srew [1] nil using solve .
continue 1 .
continue 2 .

sload bt-graphs
srew graph(5, p(0, 1) p(0, 2) p(0, 3) p(0, 4)
	p(1, 2) p(3, 4), 1) using solve .
srew graph(3, p(0, 1) p(1, 2) p(2, 0), 0) using solve .

select COLORING-BT-STRAT .
srew graph(10, 2, p(0, 1) p(0, 4) p(0, 5) p(1, 2) p(1, 6)
	p(2, 3) p(2, 7) p(3, 4) p(3, 8) p(4, 9)
	p(5, 7) p(5, 8) p(6, 8) p(6, 9) p(7, 9), nil) using solve .
srew [2] graph(10, 3, p(0, 1) p(0, 4) p(0, 5) p(1, 2) p(1, 6)
	p(2, 3) p(2, 7) p(3, 4) p(3, 8) p(4, 9)
	p(5, 7) p(5, 8) p(6, 8) p(6, 9) p(7, 9), nil) using solve .
continue 1 .

