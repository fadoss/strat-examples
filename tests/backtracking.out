==========================================
srewrite [1] in LABYRINTH-BT-STRAT : [7,7] using solve .

Solution 1
rewrites: 852
result NeList{Pos}: [7,7] [8,7] [8,8]

Solution 2
rewrites: 0
result NeList{Pos}: [7,7] [7,8] [8,8]
==========================================
srewrite [1] in QUEENS-BT-STRAT : nil using solve .

Solution 1
rewrites: 285296
result NeList{Nat}: 1 5 8 6 3 7 2 4

Solution 2
rewrites: 0
result NeList{Nat}: 1 6 8 3 7 4 2 5

Solution 3
rewrites: 0
result NeList{Nat}: 1 7 4 6 8 2 5 3

Solution 4
rewrites: 0
result NeList{Nat}: 1 7 5 8 2 4 6 3
==========================================
srewrite in HAMILTONIAN-BT-STRAT : graph(5, p(0, 1) p(0, 2) p(0, 3) p(0, 4) p(
    1, 2) p(3, 4), 1) using solve .

No solution.
rewrites: 87
==========================================
srewrite in HAMILTONIAN-BT-STRAT : graph(3, p(0, 1) p(0, 2) p(1, 2), 0) using
    solve .

Solution 1
rewrites: 43
result Graph: graph(3, nil, 0 1 2 0)

Solution 2
rewrites: 43
result Graph: graph(3, nil, 0 2 1 0)

No more solutions.
rewrites: 43
==========================================
srewrite in COLORING-BT-STRAT : graph(10, 2, p(0, 1) p(0, 4) p(0, 5) p(1, 2) p(
    1, 6) p(2, 3) p(2, 7) p(3, 4) p(3, 8) p(4, 9) p(5, 7) p(5, 8) p(6, 8) p(6,
    9) p(7, 9), nil) using solve .

No solution.
rewrites: 628
==========================================
srewrite [2] in COLORING-BT-STRAT : graph(10, 3, p(0, 1) p(0, 4) p(0, 5) p(1,
    2) p(1, 6) p(2, 3) p(2, 7) p(3, 4) p(3, 8) p(4, 9) p(5, 7) p(5, 8) p(6, 8)
    p(6, 9) p(7, 9), nil) using solve .

Solution 1
rewrites: 131517
result Graph: graph(10, 3, p(0, 1) p(0, 4) p(0, 5) p(1, 2) p(1, 6) p(2, 3) p(2,
    7) p(3, 4) p(3, 8) p(4, 9) p(5, 7) p(5, 8) p(6, 8) p(6, 9) p(7, 9), 0 1 0 1
    2 1 2 2 0 0)

Solution 2
rewrites: 131517
result Graph: graph(10, 3, p(0, 1) p(0, 4) p(0, 5) p(1, 2) p(1, 6) p(2, 3) p(2,
    7) p(3, 4) p(3, 8) p(4, 9) p(5, 7) p(5, 8) p(6, 8) p(6, 9) p(7, 9), 0 1 0 1
    2 2 2 1 0 0)

Solution 3
rewrites: 0
result Graph: graph(10, 3, p(0, 1) p(0, 4) p(0, 5) p(1, 2) p(1, 6) p(2, 3) p(2,
    7) p(3, 4) p(3, 8) p(4, 9) p(5, 7) p(5, 8) p(6, 8) p(6, 9) p(7, 9), 0 1 0 2
    1 1 2 2 0 0)
Bye.
